udata/harvest/backends/dcat.py

```diff
@@ -130,17 +139,27 @@ def parse_graph(self, url, fmt) -> List[Graph]:
                     break
             graphs.append(subgraph)
 
-            for node in subgraph.subjects(RDF.type, DCAT.Dataset):
-                id = subgraph.value(node, DCT.identifier)
-                kwargs = {'page': page}
-                self.add_item(id, **kwargs)
-                if self.max_items and len(self.job.items) >= self.max_items:
-                    # this will stop iterating on pagination
-                    url = None
+            should_stop = do(page_number, subgraph)
+            if should_stop:
+                return graphs
 
-            page += 1
+            page_number += 1
 
         return graphs
+    
+    def process_datasets(self, page_number: int, page: Graph):
```

Comment:
I think `process_datasets` (plural) may be confused with `process_dataset` (singular) for not too careful readers :p

Reply:
What do you think of 523c754d1?